File: RootTools2/src/main/java/com/rarnu/tools/neo/base/BaseActivity.java
package com.rarnu.tools.neo.base;

import android.content.res.Configuration;
import com.rarnu.tools.neo.R;

public abstract class BaseActivity extends InnerActivity {

    @Override
    public boolean getCloseCondition() {
        return getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE;
    }

    @Override
    public int getBaseLayout() {
        return R.layout.layout_replacement;
    }

    @Override
    public int getReplaceId() {
        return R.id.fReplacement;
    }
}

File: RootTools2/src/main/kotlin/com/rarnu/tools/neo/base/BaseActivity.kt
package com.rarnu.tools.neo.base

import android.content.res.Configuration
import com.rarnu.tools.neo.R

/**
 * Created by rarnu on 3/23/16.
 */
abstract class BaseActivity: InnerActivity() {

    override fun getCloseCondition(): Boolean = resources.configuration.orientation == Configuration.ORIENTATION_LANDSCAPE

    override fun getBaseLayout(): Int = R.layout.layout_replacement

    override fun getReplaceId(): Int = R.id.fReplacement

}